### PR TITLE
Skip missing algorithms in local mode

### PR DESCRIPTION
**Pull request by ale-f**
Here's a minor fix for local mode that lets the framework skip over missing algorithms. (Commit 70836ecca6c6649fe91dc678ea1b790b682cde42 added this, but only for Docker mode.)

### CODE REVIEW

File: ann_benchmarks/main.py [hunk 1]
```diff
@@ -118,14 +118,32 @@ def main():
...
+            status = algorithm_status(df)
+            # If the module was loaded but doesn't actually have a constructor of
+            # the right name, then the definition is broken
+            assert status != InstantiationStatus.NO_CONSTRUCTOR, """\
+%s.%s(%s): error: the module '%s' does not expose the named constructor""" % (df.module, df.constructor, df.arguments, df.module)
+            if status == InstantiationStatus.NO_MODULE:
+                # If the module couldn't be loaded (presumably because of a missing
+                # dependency), print a warning and remove this definition from the
+                # list of things to be run
+                print """\
```
Comment (erikbern) [inline ann_benchmarks/main.py]: should use python 3 print style here

**Comment by erikbern** [comment]
does `args.local` still work? i guess i haven't tried it for a while.

might be worth adding a unit test for it... can just add something to `.travis.yml`

**Comment by erikbern** [comment]
Travis is failing on your Python 2 print statement btw, would be great if you can fix that: https://github.com/erikbern/ann-benchmarks/pull/58

Otherwise, LGTM

**Comment by ale-f** [comment]
Done: commit c9f16addeeffecfb984df7d5da2f52caf9bbfefe gets rid of the print statement. (There's also commit 9a34a228efce82cf27c40c91d40595552afdcacc, which moves a bit of Docker-specific code that I missed into the appropriate conditional block.)

File: ann_benchmarks/main.py [hunk 2]
```diff
@@ -118,14 +110,40 @@ def main():
...
+        if args.docker_tag:
+            print('running only', args.docker_tag)
+            definitions = [d for d in definitions if d.docker_tag == args.docker_tag]
+
+        if set(d.docker_tag for d in definitions).difference(docker_tags):
+            print('not all docker images available, only:', set(docker_tags))
+            print('missing docker images:', set(d.docker_tag for d in definitions).difference(docker_tags))
+            definitions = [d for d in definitions if d.docker_tag in docker_tags]
+    else:
+        def _test(df):
```
Comment (erikbern) [inline ann_benchmarks/main.py]: seems a bit unnecessary to have this function here, but that's a fairly minor point

**Comment by erikbern** [comment]
thanks @ale-f!

**Pull request merged**